### PR TITLE
fix scatter allocation

### PR DESCRIPTION
**Pull request by psychocoderHPC**
In case pagesize bytes will be allocated and within `allocChunked` an
allocation is increased based on the waste factor to fit into a chunk
with the size of a page the locking is not working because `allocPageBased`
locking is not compatible to the locking mechanism used in
`allocChunked`.

### CODE REVIEW

File: src/include/mallocMC/creationPolicies/Scatter.hpp [hunk 1]
```diff
@@ -767,10 +767,13 @@ namespace mallocMC
             {
                 if(bytes == 0)
                     return 0;
-                // take care of padding
-                // bytes = (bytes + dataAlignment - 1) & ~(dataAlignment-1); //
-                // in alignment-policy
-                if(bytes < pagesize)
+                /* Take care of padding
+                 * bytes = (bytes + dataAlignment - 1) & ~(dataAlignment-1);
+                 * in alignment-policy.
```
Comment (sbastrakov) [inline src/include/mallocMC/creationPolicies/Scatter.hpp:772]: Are these 3 lines relevant to this part? I know this PR doesn't change it, but probably they belong somewhere else?

Comment (psychocoderHPC) [reply src/include/mallocMC/creationPolicies/Scatter.hpp:772]: I kept the equation separated from the normal text that it is better readable.

**Review by sbastrakov — APPROVED** [review]
